src/ios/CDVWKInAppBrowser.m

```diff
@@ -1180,26 +1180,40 @@ - (void)webView:(WKWebView *)theWebView didStartProvisionalNavigation:(WKNavigat
 
 - (void)webView:(WKWebView *)theWebView decidePolicyForNavigationResponse:(nonnull WKNavigationResponse *)navigationResponse decisionHandler:(nonnull void (^)(WKNavigationResponsePolicy))decisionHandler
 {
-    NSArray* downloadTypes = @[@"zip", @"rar", @"tar", @"binary"];
+    NSArray* downloadTypes = @[@"zip", @"rar", @"gz", @"tar", @"binary"];
     NSString* mimeType = navigationResponse.response.MIMEType;
+    NSUInteger filesize = navigationResponse.response.expectedContentLength;
     
     NSUInteger index = [downloadTypes indexOfObjectPassingTest: ^BOOL(id format, NSUInteger idx, BOOL *stop){
         return [[mimeType lowercaseString] containsString: [(NSString *)format lowercaseString]];
     }];
-    if(index == NSNotFound){
-        decisionHandler(WKNavigationActionPolicyAllow);
-    }else{
-        NSURLSession *session = [NSURLSession sharedSession];
-        NSURLSessionDataTask *dataTask = [session dataTaskWithURL:theWebView.URL completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-            // Todo: Send appropriate response to the JS app.            
-            /*CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:@{@"type":@"ondownload", @"response": @{ @"filename": response.suggestedFilename, @"mime": response.MIMEType, @"data": [data base64EncodedStringWithOptions :0]}}];
-            [pluginResult setKeepCallback:[NSNumber numberWithBool:YES]];
-            [self.navigationDelegate.commandDelegate sendPluginResult:pluginResult callbackId:self.navigationDelegate.callbackId];*/
-        }];
-        [dataTask resume];
-        decisionHandler(WKNavigationActionPolicyCancel);
+    if(index == NSNotFound) {
+        decisionHandler(WKNavigationResponsePolicyAllow);
+        return;
+    };
+    NSURLSession *session = [NSURLSession sharedSession];
+    NSURLSessionDataTask *dataTask = [session dataTaskWithURL:theWebView.URL completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+        NSString* savePath = [self getDownloadSavePath: response.suggestedFilename];
```

Comment:
what if there is no suggested filename?

Reply:
Not possible. It generates a default name if there is no
`Content-Disposition` header.
https://developer.apple.com/documentation/foundation/nsurlresponse/1415924-suggestedfilename?language=objc